lib/streamlit/elements/image.py

```diff
@@ -149,14 +167,40 @@ def image(
 
         """
 
-        if use_column_width == "auto" or (use_column_width is None and width is None):
-            width = WidthBehaviour.AUTO
-        elif use_column_width == "always" or use_column_width is True:
-            width = WidthBehaviour.COLUMN
-        elif width is None:
-            width = WidthBehaviour.ORIGINAL
-        elif width <= 0:
-            raise StreamlitAPIException("Image width must be positive.")
+        if use_container_width is not None and use_column_width is not None:
+            raise StreamlitAPIException(
+                "`use_container_width` and `use_column_width` cannot be set at the same time.",
+                "Please utilize `use_container_width` since `use_column_width` is deprecated.",
+            )
+
+        if use_column_width is not None:
+            show_deprecation_warning(
+                "The `use_column_width` parameter has been deprecated and will be removed "
+                "in a future release. Please utilize the `use_container_width` parameter instead."
+            )
+
+            if use_column_width == "auto":
+                width = WidthBehaviour.AUTO
+            elif use_column_width == "always" or use_column_width is True:
+                width = WidthBehaviour.COLUMN
+            elif width is None:
+                width = WidthBehaviour.ORIGINAL
+            elif width <= 0:
+                raise StreamlitAPIException("Image width must be positive.")
+
+        else:
+            # Check for non-existent params first
+            if use_container_width is None and width is None:
+                use_container_width = False
+
+            if use_container_width is True:
+                width = WidthBehaviour.MAX_IMAGE_OR_CONTAINER
+            elif use_container_width is False:
+                width = WidthBehaviour.MIN_IMAGE_OR_CONTAINER
+            elif width is None:
+                width = WidthBehaviour.MAX_IMAGE_OR_CONTAINER
```

Comment:
Is this a dead `elif` branch?

If `width` is `None` and `use_container_width` is `None`, `use_container_width` is set to `False` [in line 194](https://github.com/streamlit/streamlit/pull/9547/files#diff-4795158725eb992f73736495442ff69edd6e7ccd7609291339a61d1b6e39d716R194) and `width` gets overwritten in the `elif` branch before this check (and thus is not `None`).
Independent of the value of `width`, if `use_container_width` is not `None` (either `True` or `False`), the  `width` gets overwritten in the `if-elif` branches before this check (and thus is not `None`).
If `width` is not `None`, this branch here does not become effective anyways.

Reply:
If I am not mistaken in my previous comment, I believe the correct check would be:

```python
if use_container_width is True:
    width = WidthBehaviour.MAX_IMAGE_OR_CONTAINER
elif use_container_width is False:
    if width is None:
        width = WidthBehaviour.MIN_IMAGE_OR_CONTAINER
    elif width <= 0:
        raise StreamlitAPIException("Image width must be positive.")
    # if width > 0, width just stays the width
```

Reply:
Great comments! The dead branch is left in order to adhere to MyPy. In tandem with my above comment, I don't think we can shorten this logic due to the desire to be able to deprecate `use_column_width`, so I think leaving this as-is is the best course of action right now. Please let me know if you're seeing this differently!

Reply:
Ohh ok, damn MyPy! 😄 
Well, in this case I think I would prefer introducing a new local variable like this:

```python
image_width: int = WidthBehaviour.ORIGINAL if width is None else width
if use_column_width is not None:
    show_deprecation_warning(
        "The `use_column_width` parameter has been deprecated and will be removed "
        "in a future release. Please utilize the `use_container_width` parameter instead."
    )

    if use_column_width == "auto":
        image_width = WidthBehaviour.AUTO
    elif use_column_width == "always" or use_column_width is True:
        image_width = WidthBehaviour.COLUMN
    elif image_width <= 0:
        raise StreamlitAPIException("Image width must be positive.")

else:
    # Check for non-existent params first
    if use_container_width is None and width is None:
        use_container_width = False

    if use_container_width is True:
        image_width = WidthBehaviour.MAX_IMAGE_OR_CONTAINER
    elif use_container_width is False:
        image_width = WidthBehaviour.MIN_IMAGE_OR_CONTAINER
    elif image_width <= 0:
        raise StreamlitAPIException("Image width must be positive.")
``` 

which makes MyPy happy to get rid of the dead if branch 🙂

Reply:
Similar to my last comment, I did some iterations and the latest revision should reflect this (with a few small tweaks). Let me know what you think, thanks!